### DISCUSSION TITLE
Suggested changes to the Readme

### DISCUSSION THREAD

**Discussion by alexjalex** [Ranga-Auaha-Ako/canvas-grid-builder#80] [General]
* Include version of Python needed. Python 3 would need pip3 in cml, not pip.
* The .env setup could be less confusing by telling people to simply modify an existing .env file. Rename the .env.example to just .env.
* CANVAS_BLOCKS_THEME_CONTACT_NAME should include an actual example.
* CANVAS_BLOCKS_THEME_CONTACT_LINK should include an example. What format does this accept or want? An email? A URL? A tag for a link?
* The Beta custom icons editor (https://icon-editor.c.raa.amazon.auckland.ac.nz/) link doesn't work. Is this locked to auckland uni intranet?
* The [Canvas Design Blocks CSS](https://ranga-auaha-ako.github.io/canvas-design-blocks/theme.css) link is broken
* The [Canvas Design Blocks JS](https://ranga-auaha-ako.github.io/canvas-design-blocks/theme.js) link is broken
* The line, "All element managers should extend the ElementManager class, and be loaded in desktop.ts" is incorrect. They are loaded in EditorLoader.ts

**Comment by yhua219** [comment]
@alexjalex  Thank you very much for raising these issues and sharing your suggestions.  

### Fixed:

We have now updated **README.md** and **.env.example** to address the following issues you have raised:

- [x] Python version:  Added Python version requirements in the `Prerequisites` section.  

- [x] .env file examples:  Added explanations and examples of CANVAS_BLOCKS_THEME_CONTACT_NAME, CANVAS_BLOCKS_THEME_CONTACT_LINK in the **.env.example** file and the README `Configuration and Building` section.

- [x] Fixed the url for The Canvas Design Blocks CSS and The Canvas Design Blocks JS scripts in the `Testing the theme` section. 

- Note: These scripts were auto-built through GitHub workflow so are invisible in this repository. However, they are from the `dist` subdirectory generated by running a local build, where you can check the actual file names (i.e. url paths). The generated file names have changed at some point, hence the URL mismatch.

- [x] In the `Creating new elements` section, updated the line "All element managers should extend the ElementManager class ..." into 
 > All element managers should extend the ElementManager class, and be loaded in EditorLoader.ts to be imported to and instantiated in desktop.ts.

### Pending:
We are working on the Beta custom icons editor (https://icon-editor.c.raa.amazon.auckland.ac.nz/) link.  Will update you here as soon as it is fixed. :)

**Comment by yhua219** [comment]
@alexjalex , the  Beta custom icons editor  [https://icon-editor.c.raa.amazon.auckland.ac.nz/](https://icon-editor.c.raa.amazon.auckland.ac.nz/) is back online now. :-)
